examples/3d/render_to_texture.rs

```diff
@@ -0,0 +1,216 @@
+use bevy::{
+    core_pipeline::{draw_3d_graph, node, AlphaMask3d, Opaque3d, Transparent3d},
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        camera::{ActiveCameras, Camera, CameraProjection, ExtractedCameraNames, RenderTarget},
+        render_graph::{NodeRunError, RenderGraph, RenderGraphContext, SlotValue},
+        render_phase::RenderPhase,
+        render_resource::{
+            Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages,
+        },
+        renderer::RenderContext,
+        view::RenderLayers,
+        RenderApp, RenderStage,
+    },
+};
+
+pub const RENDER_IMAGE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Image::TYPE_UUID, 13378939762009864029);
+
+pub const FIRST_PASS_DRIVER: &str = "first_pass_driver";
+pub const FIRST_PASS_CAMERA: &str = "first_pass_camera";
+
+#[derive(Component)]
+struct FirstPassCube;
+#[derive(Component)]
+struct MainPassCube;
+
+/// rotates the inner cube (first pass)
+fn rotator_system(time: Res<Time>, mut query: Query<&mut Transform, With<FirstPassCube>>) {
+    for mut transform in query.iter_mut() {
+        transform.rotation *= Quat::from_rotation_x(1.5 * time.delta_seconds());
+        transform.rotation *= Quat::from_rotation_z(1.3 * time.delta_seconds());
+    }
+}
+
+/// rotates the outer cube (main pass)
+fn cube_rotator_system(time: Res<Time>, mut query: Query<&mut Transform, With<MainPassCube>>) {
+    for mut transform in query.iter_mut() {
+        transform.rotation *= Quat::from_rotation_x(1.0 * time.delta_seconds());
+        transform.rotation *= Quat::from_rotation_y(0.7 * time.delta_seconds());
+    }
+}
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut active_cameras: ResMut<ActiveCameras>,
+    mut images: ResMut<Assets<Image>>,
+    mut clear_color: ResMut<ClearColor>,
+) {
+    let size = Extent3d {
+        width: 512,
+        height: 512,
+        ..Default::default()
+    };
+    let image = Image {
+        data: vec![0; size.width as usize * size.height as usize * 4],
```

Comment:
Why are we multiplying our size by 4 here? (I'm pretty sure it's for the 4 RGBA channels, but that won't be obvious to everyone).

Reply:
It's the pixel size in bytes. I've eliminated this by using `Image::resize()`.